src/Microsoft.AspNetCore.SignalR.Core/HubEndPoint.cs

```diff
@@ -83,7 +83,11 @@ public class HubEndPoint<THub> where THub : Hub
             catch (Exception ex)
             {
                 Log.ErrorDispatchingHubEvent(_logger, "OnConnectedAsync", ex);
-                throw;
+
+                await SendCloseAsync(connection, ex);
+
+                // return instead of throw to let close message send successfully
```

Comment:
FYI @davidfowl this is needed because if we rethrow, it ends up completing the transport pipe with the exception. The Pipe then throws from `ReadAsync` and we can't drain remaining data (i.e. the close frame)